### PR TITLE
feat: _Quality gate fixes_

### LINKED ISSUE #20: Wrong image link in "notice for docker images"
The link to the DockerHub page of the data-exchange-test-service image defined in the `README` is wrong.
It points to https://hub.docker.com/r/tractusx/dataex. 
I think the correct link should be: https://hub.docker.com/r/tractusx/data-exchange-test-service 

blocks #19

### LINKED ISSUE #22: Copyright header missing on `Dockerfile`
There is on copyright header present on the [Dockerfile](https://github.com/eclipse-tractusx/data-exchange-test-service/blob/main/Dockerfile)

blocks #19

### LINKED ISSUE #23: Use userIds when specifying the user for the Docker image
## Description

The current [Dockerfile](https://github.com/eclipse-tractusx/data-exchange-test-service/blob/main/Dockerfile) does specify a user to run the process with, but it is only specified with name and not by ID.
Like specified in [TRG ](https://eclipse-tractusx.github.io/docs/release/trg-4/trg-4-03#implementation), the user also has to be set in the Chart deployment. Therefore, userIDs should be used in the Dockerfile as well, so the IDs can match

blocks #19

### LINKED ISSUE #24: Clarification: Dockerfile approach
I do not really understand the current approach used in the Dockerfile.
Specifically the part, where the built `.jar` file is unpackaged, and then the unpackaged files are copied to the resulting image.
As i can see from your `pom.xml`, you are creating a Spring boot based app. The most common approach other teams are using is to just build the `.jar` and do a `java -jar app.jar` without any unpacking. 

Is there a specific reason why you are using the approach you currently use?

Also: ist the `apk upgrade` step really necessary? We want to build images, that are as close to the info we provide in our Docker notice as possible. Doing an upgrade would alter the image an the info provided by the base image project is maybe out date.
So if it's not needed, please remove that command.

blocks #19

### LINKED ISSUE #29: Legal info for documentation missing
The `Arc42.md` documentation is missing the necessary legal notice section.
This section has to be "human readable", which means, it has to be added as regular Markdown section instead of a comment, like we do with copyright headers in source files.

Additional info in [TRG 7.07](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-07#documentation)

blocks #19

### LINKED ISSUE #30: Add `helm upgrade` workflow step
The current [Chart Lint and Test](https://github.com/eclipse-tractusx/data-exchange-test-service/blob/main/.github/workflows/helm-test.yaml) workflow needs to be enhanced, to additionally check the upgradeability of the product Helm Chart.
Somehow it already contains a workflow input for that (check the description though, seems like a copy-paste error), but that input ist not used, since the `helm upgrade step is missing`.

Helm upgrade itself is specified in [TRG 5.11](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-11).
An example workflow step, that can be used, can be found in [TRG 5.09](https://eclipse-tractusx.github.io/docs/release/trg-5/trg-5-09) -> last step in the example workflow

blocks #19

### LINKED ISSUE #31: `ct install` skipped on manual workflow trigger
The [Chart testing workflow](https://github.com/eclipse-tractusx/data-exchange-test-service/blob/main/.github/workflows/helm-test.yaml#L94) is skipping the `ct install` step on manual `workflow_dispatch`. This is caused by a condition, that only runs the step, if the Chart has changes to the last run.
On a manual trigger, the Chart won't have changes, since they have been released by automated workflows before.
The purpose of the `workflow_dispatch` trigger is to verify, if the Chart can be installed on a given Kubernetes version on demand, like in the current quality gate reviews.

Either the conditional on the workflow step has to be removed, or altered in a way, that it is always run on `workflow_dispatch` events

Manual tests could be done this time, so not necessarily blocking #19

### PR DESCRIPTION
**Pull request by adkumar1**
- This PR includes fixes for issues in QG
- Fixes #20 , Fixes #22 , Fixes #23 , Fixes #24 , Fixes #29 , Fixes #30 , Fixes #31

### CODE REVIEW

File: .github/workflows/helm-test.yaml [hunk 1]
```diff
@@ -28,7 +28,6 @@ on:
   pull_request:
     paths:
       - 'charts/**'
-  workflow_dispatch:
```
Comment (SebastianBezold) [inline .github/workflows/helm-test.yaml:31]: I think this is removed by accident

File: .github/workflows/helm-test.yaml [hunk 2]
```diff
@@ -39,7 +38,7 @@ on:
       upgrade_from:
         description: 'portal chart version to upgrade from'
```
Comment (SebastianBezold) [inline .github/workflows/helm-test.yaml]: This comment is still wrong

File: .github/workflows/helm-test.yaml [hunk 3]
```diff
@@ -94,3 +93,11 @@ jobs:
       - name: Run chart-testing (install)
         run: ct install --charts charts/data-exchange --config charts/chart-testing-config.yaml --helm-extra-set-args "--set=image.tag=testing --set=image.repository=kind-registry:5000/data-exchange-test-service"
         if: ${{ env.CHART_CHANGED == 'true' }}
+              # Upgrade the released chart version with the locally available chart
```
Comment (SebastianBezold) [inline .github/workflows/helm-test.yaml]: The indentation seems to be too deep here

**Review by SebastianBezold — CHANGES_REQUESTED** [review]
Hi @adkumar1, thanks for the fixes. Only minor issues in the helm-test workflow, that you should fix before merging

File: .github/workflows/helm-test.yaml (hunk 1 above)
Comment (adkumar1) [reply .github/workflows/helm-test.yaml:31]: Corrected.

File: .github/workflows/helm-test.yaml (hunk 2 above)
Comment (adkumar1) [reply .github/workflows/helm-test.yaml]: Updated

File: .github/workflows/helm-test.yaml (hunk 3 above)
Comment (adkumar1) [reply .github/workflows/helm-test.yaml]: Corrected.

**Review by SebastianBezold — APPROVED** [review]

**Pull request merged**